src/screens/crew6/relayScreen.cpp

```diff
@@ -162,6 +163,17 @@ RelayScreen::RelayScreen(GuiContainer* owner, bool allow_comms)
     });
     launch_probe_button->setSize(GuiElement::GuiSizeMax, 50)->setVisible(my_spaceship.hasComponent<ScanProbeLauncher>());
 
+    // Center on ship
+    center_button = new GuiButton(option_buttons, "CENTER_ON_SHIP", tr("Center On Ship"), [this](){
+        if(!my_spaceship) return;
+        auto transform = my_spaceship.getComponent<sp::Transform>();
+        if(!transform) // If my ship has no position, it must be docked inside another ship
+            transform = my_spaceship.getComponent<DockingPort>()->target.getComponent<sp::Transform>();
```

Comment:
in theory, a ship could have no transform and no docking port. So docking port also needs to be checked for nullptr.
```
if (!transform) {
  if (auto dp = my_spaceship.getComponent<DockingPort>()) {
    transform = dp->target.getComponent<sp::Transform>();
  }
}
```